feat: Add Elastic Beanstalk MCP server implementation

amjadsy opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
## Summary
Elastic Beanstalk MCP server implementation. Adds support for 4 Describe* Operations in Elastic Beanstalk Allowing customers to chat with an LLM about their Beanstalk environments.

### Changes

Adding a new MCP server for Elastic Beanstalk that will allow users to chat with the LLM about their environments, applications, configuration settings and more.

### User experience

The user can choose to incorporate the MCP server into an LLM platform of their choice and receive the functionality.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [Yes] I have reviewed the [contributing guidelines](https://github.com/awslabs/mcp/blob/main/CONTRIBUTING.md)
* [Yes] I have performed a self-review of this change
* [Yes] Changes have been tested
* [Will be formally published by awslabs team] Changes are documented

Is this a breaking change? (N)

**RFC issue number**: 446

Checklist:

* [Will be formally published by awslabs team ] Migration process documented
* [Yes, error handling implemented] Implement warnings (if it can live side by side)

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the [project license](https://github.com/awslabs/mcp/blob/main/LICENSE).

alexa-perlov commented:
Thank you for the contribution! Please take a look at the failing tests

krokoko commented:
Hi @amjadsy , just checking if you had a chance to look at the PR comments ? Thanks !

amjadsy commented:
@krokoko I've addressed your comments, please have a look and let me know if everything looks good, Thanks

@alexa-perlov All the tests should be passing now, please let me know if we're good to publish. Thanks

krokoko commented on src/elasticbeanstalk-mcp-server/awslabs/elasticbeanstalk_mcp_server/consts.py:
```diff
@@ -0,0 +1,33 @@
+DEFAULT_MAX_RECORDS = 100
+
+# API configuration
+USER_AGENT_EXTRA = 'MCP/BeanstalkServer'
```
Could you please update the user agent extra to follow the structure `AWSLABS/MCP/{MCP_SERVER_NAME}/{MCP_SERVER_VERSION}`

krokoko commented on src/elasticbeanstalk-mcp-server/README.md:
```diff
@@ -0,0 +1,137 @@
+# AWS Elastic Beanstalk MCP Server
```
could you please add to the readme a cursor deeplink ? You can have a look at other servers for examples
